lib/ancestry/has_ancestry.rb

```diff
@@ -30,6 +30,8 @@ def has_ancestry options = {}
       validates_format_of self.ancestry_column, :with => derive_ancestry_pattern(options[:primary_key_format]), :allow_nil => true
       extend Ancestry::MaterializedPath
 
+      include Ancestry::MaterializedPathPg if Ancestry.optimizer
```

Comment:
Yea, I'm still leaning towards this.
Please share where I flubbed it

```suggestion
      include Ancestry::MaterializedPathPg if connection.adapter_name.downcase == "postgresql"
```

Think this is what we do in other places.
If connection can not be accessed, it may be `class.connection` or something like that? Pretty sure `self` is the class coming in.

Reply:
If you think optimization should be opt in, then you add an opt-in parameter (`:sql_updates =>true`)

I do feel it is model specific than database specific.
Just like `:dependent => :destroy` and `:dependent => :delete`